### PR TITLE
Code Review for [create-model-db] | DevThoughts Blog Rails Project

### PR DESCRIPTION
**Pull request by luigirazum**
# Hello Dear Code Reviewers Team 👋,
### - 🎯 General requirements
- [X] Make sure that there are _**no** linter errors_.
  - 🟢 Linters for `rubocop` and `stylelint` were installed following the [RoR linters guide](https://github.com/microverseinc/linters-config/blob/master/ror).
  - 🟢 `rubocop` ran and **no offenses** were detected.
- [X] Make sure that you used _**correct** flow_ `git-flow`.
  - 🟢 The `dev` branch was created from `main`.
  - 🟢 Then the `dev` branch was set as `default` branch.
  - 🟢 The <kbd>Milestone</kbd> branch `create-model-db` was created from `dev` to follow the `git-flow`.
- [X] Make sure that you documented your work _in a **professional way**_.
  - 🟢 All the information about the project was added to the `README.md` file.

### - 🎯 Project requirements
- [X] Initialize your project with Git.
  - 🟢 A new repo [`devthoughts-blog`](https://github.com/luigirazum/devthoughts-blog) was created at GitHub and then cloned locally.
- [X] Create a new app. Give it a meaningful name.
  - 🟢 <kbd>DevThoughts Blog</kbd> App was created.
- [X] Make sure that your project has Postgres database set up.
  - 🟢 When the App was created, the option `--database=postgresql` was used.
- [X] Build your project schema.
  - 🟢 A `schema.rb` was generated to configure our database.
  - [X] Create and run the necessary migration files.
    - 🟢 `CreateUsers` migration was generated to create `users` table. 
    - 🟢 `CreatePosts` migration was generated to create `posts` table.
    - 🟢 `CreateComments` migration was generated to create `comments` table.
    - 🟢 `CreateLikes` migration was generated to create `likes` table.
  - [X] Table and column names should match the ERD diagram.
    _**Note:** photo for users table should be a link to an image._
    - 🟢 All column names match the `ERD Diagram`
    - 🟢 `photo` column at `users` table is `string` to store a link.
  - [X] Foreign keys should be included.
    - 🟢 All FK that reference another table are included.
    - [X] Pay attention that in `posts` table the column holding the foreign key to the `users` table should be named `author_id`
      - 🟢 `author_id` references `id` at `users` table.
  - [X] All columns that are foreign keys should have a corresponding index.
    - 🟢 An INDEX was created for all FK columns.

### - 🎯 ERD Diagram
- The database `schema` was created according to the following **ERD diagram** 👇, that includes `table` and `column` names as well as data `relationships`.
  blog_app_erd_v1_1

### Luis 🙋‍♂️

### CODE REVIEW

File: db/migrate/20230927141844_add_author_foreign_key_to_posts.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class AddAuthorForeignKeyToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_column :posts, :author_id, :bigint
+    add_foreign_key :posts, :users, column: :author_id, primary_key: 'id', index: true
```
Comment (rloterh) [inline db/migrate/20230927141844_add_author_foreign_key_to_posts.rb:4]: - [ ] `Author` is aliasing `User` model and thus we have `AuthorId` used as reference for associated models to `User`, as seen in the given [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). Hence, our migration query should be similar to:

```ruby
...
  def change
    add_reference :posts, :author, references: :users, null: false, foreign_key: {to_table: :users}
  end
...

```

or.. 

```ruby
  create_table :posts do |t|
    . . .
    t.references :author, foreign_key: { to_table: 'users' } 
    . . .
  end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Luis, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦 🌴  🏃 🌳 🏡 🏘️ 🌲 


## Highlights 👍  
- Good job so far on establishing migration queries!
- ReadMe looks great

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230927141844_add_author_foreign_key_to_posts.rb (hunk 1 above)
Comment (luigirazum) [reply db/migrate/20230927141844_add_author_foreign_key_to_posts.rb:4]: Thanks for the suggestion @rloterh.
✅ Applied

**Review by DammyShittu — APPROVED** [review]
Hi @luigirazum,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**